src/core_plugins/kibana/public/context/components/size_picker/size_picker.js

```diff
@@ -0,0 +1,30 @@
+import _ from 'lodash';
+import uiModules from 'ui/modules';
+import contextSizePickerTemplate from './size_picker.html';
+import './size_picker.less';
+
+const module = uiModules.get('apps/context', [
+  'kibana',
+]);
+
+module.directive('contextSizePicker', function ContextSizePicker() {
+  return {
+    bindToController: true,
+    controller: ContextSizePickerController,
+    controllerAs: 'contextSizePicker',
+    replace: true,
+    restrict: 'E',
+    scope: {
+      count: '=',
+      setCount: '=',
+    },
+    template: contextSizePickerTemplate,
+  };
+});
+
+
+function ContextSizePickerController() {
+  this.getOrSetCount = (count) => (
+    _.isUndefined(count) ? this.count : this.setCount(count)
```

Comment:
I guess I'm just worried that providing a setter for ng-model that doesn't actually set anything could cause subtle bugs with angular's two way binding. The validity of this component's internal state is essentially dependent upon implementation details of the parent component's `setCount` callback.

The main issue is that we can't one way bind the directive's `count` property. What if we did something similar to what you've already done in the context app controller? Create a scope variable called `state.count`. Watch the `count` property and update `state.count` when it changes. Bind `state.count` with ng-model on the input element. Watch for changes in `state.count`, fire `setCount` when it changes if `state.count !== count`. 

This might be a useful pattern to establish to accomplish "one way" binding until we upgrade angular.